Avoid Number instance in `utxo-lib` output sorting

komret opened the pull request:
## Description

5b614b52c0d23db808ec96c237c1776bad6a1eae fixes bug reported on [Slack](https://satoshilabs.slack.com/archives/CKZHV2G13/p1668949032736539). It happened when sorting outputs in `utxo-lib` where subtracting one output amount from another gave a result higher than MAX_SAFE_INTEGER. Fixed by substituting subtraction with `bn.cmp()` function.

The other commits are tests that helped discover the problem. They could be handy in future as well:
a871569fb142b0d1e8bb9ecb28083699e5e00bf1 enables running connect tests on Mac.
011bd8f5a4a483ecc3ea9977ab062b02d260244c adds a `composeTransaction` method test.
b6258eabce6bae7963481164acc14fb79c007667 adds an e2e test for Dogecoin with mocked blockbook backend and fixes the path in README to start the test suite.

**QA:** This cannot be tested manually unless you are a Dogecoin whale, hence the mocked responses in b6258eabce6bae7963481164acc14fb79c007667 which increase the account balance. This PR does not introduce any behaviour change.

szymonlesisz commented:
👌 perfect, turns out that `compose` tests in utxo-lib didn't covered that part :/ 
could you add some DOGE fixtures to `utxo-lib/tests/__fixtures__/compose` as well?

komret commented:
@szymon, I moved the test from `connect` to `utxo-lib` in e718c3c60132a9f56ffe18caf69320c874c92da5, please check it out. I added `doge` to `networks.ts` rather than modifying `getNetwork()` because I find it cleaner this way. I needed not only different `bip32` than `bitcoin` has, but also `pubKeyHash` which is checked in `toOutputScript()`.

szymonlesisz commented on packages/connect/e2e/run.ts:
```diff
@@ -16,8 +16,11 @@ const firmwareModel = process.env.TESTS_FIRMWARE_MODEL;
 const getEmulatorOptions = (availableFirmwares: Firmwares) => {
-    const latest2 = availableFirmwares['2'].find(fw => !fw.includes('-'));
-    const latest1 = availableFirmwares['1'].find(fw => !fw.includes('-'));
+    const getLatestFirmware = (model: keyof Firmwares) =>
```
not sure here, wouldnt it pick arm emulator for not arm users? (both are present in user env)

komret replied:
I think the emulator serves only one firmware version based on the system, so non-arm Macs shouldn't be affected. Discussed with @mroz.

szymonlesisz reviewed (approved):
utxo-lib changes looks good 👍 not sure about user-env changes
